src/software/ai/hl/stp/play/example_play.cpp

```diff
@@ -20,12 +20,13 @@ bool ExamplePlay::invariantHolds(const World &world) const
 void ExamplePlay::getNextTactics(TacticCoroutine::push_type &yield, const World &world)
 {
     // Create MoveTactics that will loop forever
-    auto move_tactic_1 = std::make_shared<MoveTactic>(true);
-    auto move_tactic_2 = std::make_shared<MoveTactic>(true);
-    auto move_tactic_3 = std::make_shared<MoveTactic>(true);
-    auto move_tactic_4 = std::make_shared<MoveTactic>(true);
-    auto move_tactic_5 = std::make_shared<MoveTactic>(true);
-    auto move_tactic_6 = std::make_shared<MoveTactic>(true);
+    std::vector<std::shared_ptr<MoveTactic>> move_tactics = {
+        std::make_shared<MoveTactic>(true), std::make_shared<MoveTactic>(true),
```

Comment:
a fill contructor should work here http://www.cplusplus.com/reference/vector/vector/vector/

Reply:
very neat, didn't know this was a thing, updated

Reply:
this doesn't work because it fills it with the same value

switched to `std::generate`